The job **`tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/20105985933/job/57689945616) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/57689945616) [(plain)](https://github.com/rust-lang/rust/commit/01f8a22360168cf2f9a3e011a369711b4edacb85/checks/57689945616/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
linting javascript files and applying suggestions
Running eslint on rustdoc JS files

/checkout/src/librustdoc/html/static/js/main.js
  914:43  error  Strings must use doublequote  quotes
  917:52  error  Strings must use doublequote  quotes

✖ 2 problems (2 errors, 0 warnings)
  2 errors and 0 warnings potentially fixable with the `--fix` option.

rerun tidy with `--extra-checks=js:lint --bless` to apply eslint suggestions
tidy [extra_checks]: checks with external tool 'eslint' failed
tidy [extra_checks]: FAIL
tidy: The following check failed: extra_checks
Bootstrap failed while executing `test src/tools/tidy tidyselftest --extra-checks=py,cpp,js,spellcheck`
Command `/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools-bin/rust-tidy /checkout /checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo /checkout/obj/build 4 /node/bin/yarn --extra-checks=py,cpp,js,spellcheck` failed with exit code 1
Created at: src/bootstrap/src/core/build_steps/tool.rs:1612:23
Executed at: src/bootstrap/src/core/build_steps/test.rs:1320:29

Command has failed. Rerun with -v to see more details.
Build completed unsuccessfully in 0:02:50
  local time: Wed Dec 10 16:40:53 UTC 2025
  network time: Wed, 10 Dec 2025 16:40:53 GMT
##[error]Process completed with exit code 1.
##[group]Run echo "disk usage:"
```

</details>

--- a/src/librustdoc/html/render/print_item.rs
+++ b/src/librustdoc/html/render/print_item.rs
@@ -650,18 +650,18 @@ fn item_function(cx: &Context<'_>, it: &clean::Item, f: &clean::Function) -> imp
 ///
 /// This marker appears once in all trait impl lists to divide negative impls from positive impls.
 struct NegativeMarker {
-    inserted_negative_marker: bool,
+    inserted: bool,
 }
 
 impl NegativeMarker {
     fn new() -> Self {
-        Self { inserted_negative_marker: false }
+        Self { inserted: false }
     }
 
     fn insert_if_needed(&mut self, w: &mut fmt::Formatter<'_>, implementor: &Impl) -> fmt::Result {
-        if !self.inserted_negative_marker && !implementor.is_negative_trait_impl() {
-            write!(w, "<div class=\"negative-marker\"></div>")?;
-            self.inserted_negative_marker = true;
+        if !self.inserted && !implementor.is_negative_trait_impl() {
+            w.write_str("<div class=\"negative-marker\"></div>")?;
+            self.inserted = true;
         }
         Ok(())
     }

--- a/src/librustdoc/html/render/write_shared.rs
+++ b/src/librustdoc/html/render/write_shared.rs
@@ -747,7 +747,14 @@ impl TraitAliasPart {
             path.push(format!("{remote_item_type}.{}.js", remote_path[remote_path.len() - 1]));
 
             let mut implementors = implementors.collect::<Vec<_>>();
-            implementors.sort_unstable();
+            implementors.sort_unstable_by(|a, b| {
+                // We sort negative impls first.
+                match (a.is_negative, b.is_negative) {
+                    (false, true) => Ordering::Greater,
+                    (true, false) => Ordering::Less,
+                    _ => compare_names(&a.text, &b.text),
+                }
+            });
 
             let part = OrderedJson::array_unsorted(
                 implementors
@@ -762,7 +769,6 @@ impl TraitAliasPart {
     }
 }
 
-#[derive(Eq)]
 struct Implementor {
     text: String,
     synthetic: bool,
@@ -783,29 +789,6 @@ impl Serialize for Implementor {
             seq.serialize_element(&self.types)?;
         }
         seq.end()
-    }
-}
-
-impl PartialEq for Implementor {
-    fn eq(&self, other: &Self) -> bool {
-        self.cmp(other) == Ordering::Equal
-    }
-}
-
-impl PartialOrd for Implementor {
-    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
-        Some(Ord::cmp(self, other))
-    }
-}
-
-impl Ord for Implementor {
-    fn cmp(&self, other: &Self) -> Ordering {
-        // We sort negative impls first.
-        match (self.is_negative, other.is_negative) {
-            (false, true) => Ordering::Greater,
-            (true, false) => Ordering::Less,
-            _ => compare_names(&self.text, &other.text),
-        }
     }
 }
 

--- a/src/librustdoc/html/static/css/noscript.css
+++ b/src/librustdoc/html/static/css/noscript.css
@@ -29,7 +29,7 @@ nav.sub {
 	display: none;
 }
 
-#synthetic-implementors-list, #implementors-list {
+#synthetic-implementors-list:not(.loaded), #implementors-list:not(.loaded) {
 	display: block;
 }
 

--- a/src/librustdoc/html/static/css/rustdoc.css
+++ b/src/librustdoc/html/static/css/rustdoc.css
@@ -1158,7 +1158,7 @@ div.where {
 	margin-left: calc(var(--docblock-indent) + var(--impl-items-indent));
 }
 
-#synthetic-implementors-list, #implementors-list {
+#synthetic-implementors-list:not(.loaded), #implementors-list:not(.loaded) {
 	/* To prevent layout shift when loading the page with extra implementors being loaded
 	   from JS, we hide the list until it's complete. */
 	display: none;

--- a/src/librustdoc/html/static/js/main.js
+++ b/src/librustdoc/html/static/js/main.js
@@ -911,10 +911,10 @@ function preLoadCss(cssUrl) {
             }
         }
         if (implementors[0]) {
-            implementors[0].style.display = "block";
+            implementors[0].classList.add("loaded");
         }
         if (syntheticImplementors[0]) {
-            syntheticImplementors[0].style.display = "block";
+            syntheticImplementors[0].classList.add("loaded");
         }
     };
     if (window.pending_implementors) {

